### ISSUE TITLE
Add support of the `test` split on ImageNet

### ISSUE THREAD

**Issue by datumbox** [pytorch/vision#4855]
### 🚀 The feature

The new Dataset API allows returning meta-data about the images. This means that the datasets can also be used for producing predictions. As a result now it makes sense for the `ImageNet` dataset to also support the `test` split which doesn't contain labels but instead is a flat list of files.

### Motivation, pitch

See this discussion for more info: https://github.com/pytorch/vision/pull/4845#discussion_r742055625

cc @pmeier @bjuncek

**Comment by pmeier** [comment]
Unfortunately, I don't have access to the test data, so I need more information to add this. 

- Train and validation split come in archives:
    https://github.com/pytorch/vision/blob/b47ea8340c84bbce8fc3107034fd218751f007a0/torchvision/prototype/datasets/_builtin/imagenet.py#L57-L66
    Following the pattern, is the test archive `ILVSRC2012_img_test.tar`?
- If yes, is it modeled like the validation archive? If yes, the images should be directly in the archive and probably named `ILSVRC2012_test_00000001.JPEG`.

**Comment by datumbox** [comment]
Yes there is a `ILVSRC2012_img_test.tar`. See these [old instructions](https://blog.artwolf.in/a?ID=d3458c31-985f-4fd6-b07a-f103cb318adf) to get a better understanding on its structure.

You can get access to the archieve by [signing up](https://image-net.org/signup.php) for a research access and then download the data from [here](https://image-net.org/challenges/LSVRC/2012/2012-downloads.php).

Note that the test tar doesn't have labels.
